internal/provider/resource_gitlab_system_hook_test.go

```diff
@@ -0,0 +1,127 @@
+package provider
+
+import (
+	"fmt"
+	"strconv"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/xanzy/go-gitlab"
+)
+
+func TestAccGitlabSystemHook_basic(t *testing.T) {
+	var hook gitlab.Hook
+	rInt := acctest.RandInt()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: providerFactories,
+		CheckDestroy:      testAccCheckGitlabSystemHookDestroy,
+		Steps: []resource.TestStep{
+			// Create a hook with all options
+			{
+				Config: testAccGitlabSystemHookConfig(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGitlabSystemHookExists("gitlab_system_hook.this", &hook),
+					resource.TestCheckResourceAttrSet("gitlab_system_hook.this", "created_at"),
+				),
+			},
+			// Verify import
+			{
+				ResourceName:            "gitlab_system_hook.this",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"token"},
+			},
+			// Update the hook to toggle all the values to their inverse
+			{
+				Config: testAccGitlabSystemHookUpdateConfig(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGitlabSystemHookExists("gitlab_system_hook.this", &hook),
+				),
```

Comment:
@timofurrer  - tiny nitpick, but since this will force the creation of a new resource, do we want to logically check that the "created_at" date changed in this test as opposed to line 28?

Otherwise this looks good to me!

Reply:
hmmm, the reason I didn't is that from an acceptance test perspective I actually don't really care. Even if an update endpoint would be implemented one day, the essence of the feature still remains the same and so the tests.

On the other side it makes sense to test ALL the current behavior, but I'd mark this an implementation detail and I think we are good not asserting it in the acceptance tests.

Reply:
That's fair. I'd say this to good to approve then.